### PR TITLE
feat(RedPacket): mf-6048 custom cover

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/mf-6048
https://mask.atlassian.net/browse/mf-6514

### CODE REVIEW

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useHandleCreateOrSelect.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    )
+                    payload.password = payload.password.slice(2)
+                }
+            }
+
+            if (!isFirefly && senderName) {
+                payload.sender.name === senderName
+            }
+
+            const post = t`Hi friends, I just created a token Lucky Drop. Download mask.io to claim. Follow @realMaskNetwork for Web3 updates and insights.
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/hooks/useHandleCreateOrSelect.ts:55]: https://github.com/DimensionDev/firefly.mask.social/pull/3489/files

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useMyCollectionNfts.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        chainId,
+        size: 50,
+    })
+    const { data: assets_ = EMPTY_LIST, hasNextPage, fetchNextPage } = result
+
+    const assets = collection?.assets?.length ? collection.assets : assets_
+    const nfts = useMemo(() => assets.map((v, index) => ({ ...v, index }) as OrderedERC721Token), [assets])
+
+    useEffect(() => {
+        fetchNextPage()
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/hooks/useMyCollectionNfts.ts]:
```suggestion
        if (hasNextPage) fetchNextPage()
```

File: packages/plugins/RedPacket/src/SiteAdaptor/views/CreateERC20RedPacket.tsx [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+    )
+
+    // amount
+    const amount = rightShift(rawAmount || '0', token?.decimals)
+    const rawTotalAmount = useMemo(
+        () => (isRandom || !rawAmount ? rawAmount : multipliedBy(rawAmount, shares).toFixed()),
+        [rawAmount, isRandom, shares],
+    )
+    const totalAmount = useMemo(() => multipliedBy(amount, isRandom ? 1 : (shares ?? '0')), [amount, shares, isRandom])
+    const minTotalAmount = useMemo(() => new BigNumber(isRandom ? 1 : (shares ?? 0)), [shares, isRandom])
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/views/CreateERC20RedPacket.tsx]: Remove memos

File: packages/plugins/RedPacket/src/SiteAdaptor/views/CreateERC20RedPacket.tsx [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+
+    const gasValidationMessage = (() => {
+        if (!token) return ''
+        if (!isGasSufficient) {
+            return <Trans>Insufficient Balance for Gas Fee</Trans>
+        }
+        if (!loadingTransactionValue && new BigNumber(transactionValue).isLessThanOrEqualTo(0))
+            return <Trans>Insufficient Balance</Trans>
+
+        return undefined
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/views/CreateERC20RedPacket.tsx]:
```suggestion
        if (!token) return
        if (!isGasSufficient) {
            return <Trans>Insufficient Balance for Gas Fee</Trans>
        }
        if (!loadingTransactionValue && new BigNumber(transactionValue).isLessThanOrEqualTo(0))
            return <Trans>Insufficient Balance</Trans>

        return
```

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useMyCollectionNfts.ts (hunk 2 above)
Comment (swkatmask) [reply packages/plugins/RedPacket/src/SiteAdaptor/hooks/useMyCollectionNfts.ts]: Nice catch.

File: packages/plugins/RedPacket/src/SiteAdaptor/views/HistoryDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                        chainId,
+                        createIndicator(undefined, pageParam as string),
+                    )
+                    return res
+                }
+            :   skipToken,
+        getNextPageParam: (lastPage) => lastPage.cursor,
+        // select(data) {
+        //     return data.pages.flatMap((x) => x.list)
+        // },
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/views/HistoryDetail.tsx]:
```suggestion
```

File: packages/plugins/RedPacket/src/SiteAdaptor/views/HistoryDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    const history = location.state.history as FireflyRedPacketAPI.RedPacketSentInfo
+    const [params] = useSearchParams()
+    const rpid = params.get('id')
+    const rawChainId = params.get('chain-id')
+    const chainId = rawChainId ? +rawChainId : undefined
+
+    const { data, isLoading } = useInfiniteQuery({
+        enabled: !!rpid,
+        queryKey: ['redpacket', 'claim-history', rpid, chainId],
+        initialPageParam: undefined as any,
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/views/HistoryDetail.tsx]: Avoid any.

**Review by guanbinrui — DISMISSED** [review]

**Pull request merged**